pkg/streamingpromql/types/limiting_pool_test.go

```diff
@@ -190,6 +192,39 @@ func TestLimitingPool_ClearsReturnedSlices(t *testing.T) {
 	})
 }
 
+func TestLimitingPool_Mangling(t *testing.T) {
+	defer func() {
+		// Ensure we reset this back to the default state given it applies globally.
+		EnableManglingReturnedSlices = false
+	}()
```

Comment:
(nit) In case we change where this is turned on in the future etc.

```suggestion
	currentEnableManglingReturnedSlices := EnableManglingReturnedSlices
	defer func() {
		// Ensure we reset this back to the default state given it applies globally.
		EnableManglingReturnedSlices = currentEnableManglingReturnedSlices
	}()
	EnableManglingReturnedSlices = false
```

Reply:
Added in 7293a8b.